cluster-autoscaler/core/scale_up.go

```diff
@@ -367,27 +370,36 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 			Pods:      make([]*apiv1.Pod, 0),
 		}
 
-		schedulableOnNode := CheckPodsSchedulableOnNode(context, unschedulablePods, nodeGroup.Id(), nodeInfo)
-		for pod, err := range schedulableOnNode {
-			if err != nil {
+		// add list of pods which pass predicates to option
+		podsPassing, err := getPodsPassingPredicates(nodeGroup.Id())
+		if err != nil {
+			glog.V(4).Infof("Skipping node group %s; cannot compute pods passing predicates", nodeGroup.Id())
+			continue
+		} else {
+			option.Pods = make([]*apiv1.Pod, len(podsPassing))
+			copy(option.Pods, podsPassing)
+		}
+
+		// mark that there is a scheduling option for pods which can be scheduled to node from currently analyzed node group
+		for _, pod := range podsPassing {
+			delete(podsRemainUnschedulable, pod)
+		}
+
+		// update information why we cannot schedule pods for which we did not find a working extension option so far
+		podsNotPassing, err := getPodsNotPassingPredicates(nodeGroup.Id())
+		if err != nil {
```

Comment:
Since we exit on this error without saving expansion option for this node group, we should do it before deleting pod from "podsRemainUnschedulable".